x-pack/plugins/secops/public/components/timeline/footer/index.tsx

```diff
@@ -151,15 +152,29 @@ export class Footer extends React.PureComponent<FooterProps, FooterState> {
   public readonly state = {
     isPopoverOpen: false,
     paginationLoading: false,
+    updatedAt: null,
   };
 
   public componentDidUpdate(prevProps: FooterProps) {
-    const { paginationLoading } = this.state;
-    const { isLoading } = this.props;
+    const { paginationLoading, updatedAt } = this.state;
+    const { isLoading, getUpdatedAt } = this.props;
     if (paginationLoading && prevProps.isLoading && !isLoading) {
       this.setState({
         ...this.state,
         paginationLoading: false,
+        updatedAt: getUpdatedAt(),
+      });
+    }
+    if (updatedAt === null) {
+      this.setState({
+        ...this.state,
+        updatedAt: getUpdatedAt(),
+      });
+    }
+    if (prevProps.isLoading && !isLoading) {
```

Comment:
If this is same code is intended, wouldn't you combine the code into one block with one if statement?

```ts
if (updated === null || (prevProps. isLoading && !isLoading)) {
      this.setState({
        ...this.state,
        updatedAt: getUpdatedAt(),
}
```

Otherwise it reads a bit weird to me that we split on one boolean but used two below it?

Reply:
it will be part of the refactor, that we talked in an early review